File: doctor-clinic-api/src/main/kotlin/pl/monochromatic/dolphins/doctorclinic/models/dao/MedicalInterview.kt
package pl.monochromatic.dolphins.doctorclinic.models.dao

import com.fasterxml.jackson.annotation.JsonBackReference
import javax.persistence.*

@Entity
@Table(name = "medical_interviews")
class MedicalInterview(

    @Id
    @Column(name = "patient_id")
    val id: Int?,

    @OneToOne
    @MapsId
    @JsonBackReference
    @JoinColumn(name = "patient_id")
    val patient: User?,

    var pregnancy: Int? = null,
    var glucose: Int? = null,
    var bloodPressure: Int? = null,
    var skinThickness: Int? = null,
    var insulin: Int? = null,
    var bmi: Double? = null,
    var diabetesPedigree: Double? = null,
    var age: Int? = null,

    @ManyToOne
    @JoinColumn(name = "doctor_id", nullable = true)
    var doctor: User? = null
)

File: doctor_clinic_front/lib/model/medical_interview.dart
import 'package:doctor_clinic/model/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'medical_interview.g.dart';

@JsonSerializable()
class MedicalInterview {
  final int? id;
  final User? patient;
  final User? doctor;
  final int? pregnancy;
  final int? glucose;
  final int? bloodPressure;
  final int? skinThickness;
  final int? insulin;
  final double? bmi;
  final double? diabetesPedigree;
  final int? age;

  MedicalInterview({
    this.id,
    this.patient,
    this.doctor,
    this.pregnancy,
    this.glucose,
    this.bloodPressure,
    this.skinThickness,
    this.insulin,
    this.bmi,
    this.diabetesPedigree,
    this.age,
  });

  MedicalInterview copyWith({
    int? id,
    User? patient,
    User? doctor,
    int? pregnancy,
    int? glucose,
    int? bloodPressure,
    int? skinThickness,
    int? insulin,
    double? bmi,
    double? diabetesPedigree,
    int? age,
  }) =>
      MedicalInterview(
        id: id ?? this.id,
        patient: patient ?? this.patient,
        doctor: doctor ?? this.doctor,
        pregnancy: pregnancy ?? this.pregnancy,
        glucose: glucose ?? this.glucose,
        bloodPressure: bloodPressure ?? this.bloodPressure,
        skinThickness: skinThickness ?? this.skinThickness,
        insulin: insulin ?? this.insulin,
        bmi: bmi ?? this.bmi,
        diabetesPedigree: diabetesPedigree ?? this.diabetesPedigree,
        age: age ?? this.age,
      );

  factory MedicalInterview.fromJson(Map<String, dynamic> json) => _$MedicalInterviewFromJson(json);

  Map<String, dynamic> toJson() => _$MedicalInterviewToJson(this);
}
